sdk/communication/azure-communication-sms/tests/test_sms_client_e2e.py

```diff
@@ -6,100 +6,161 @@
         
-        assert len(sms_responses) is 2
+        assert len(sms_responses) == 2
 
-        for sms_response in sms_responses:
-            self.verify_sms_response(sms_response)
+        self.verify_successful_sms_response(sms_responses[0])
+        self.verify_successful_sms_response(sms_responses[1])
     
-    @pytest.mark.live_test_only
-    def test_send_sms_invalid_to_phone_number(self):
+    def test_send_sms_from_managed_identity(self):
+        endpoint, access_key = parse_connection_str(self.connection_str)
+        from devtools_testutils import is_live
+        if not is_live():
+            credential = FakeTokenCredential()
+        else:
+            credential = DefaultAzureCredential()
+        sms_client = SmsClient(endpoint, credential)
 
         # calling send() with sms values
-        sms_responses = self.sms_client.send(
+        sms_responses = sms_client.send(
             from_=self.phone_number,
-            to=["+1234567891011"],
-            message="Hello World via SMS",
-            enable_delivery_report=True,  # optional property
-            tag="custom-tag")  # optional property
+            to=[self.phone_number],
+            message="Hello World via SMS")
+        
+        assert len(sms_responses) == 1
+
+        self.verify_successful_sms_response(sms_responses[0])
+
+    def test_send_sms_fake_from_phone_number(self):
+
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+        
+        with pytest.raises(HttpResponseError) as ex:
+            # calling send() with sms values
+            sms_client.send(
+                from_="+15550000000",
+                to=[self.phone_number],
+                message="Hello World via SMS")
+        
+        assert str(ex.value.status_code) == "400"
+        assert ex.value.message is not None
+
+    def test_send_sms_fake_to_phone_number(self):
+
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+
+        # calling send() with sms values
+        sms_responses = sms_client.send(
+            from_=self.phone_number,
+            to=["+15550000000"],
+            message="Hello World via SMS")
         
-        assert len(sms_responses) is 1
+        assert len(sms_responses) == 1
 
-        for sms_response in sms_responses:
-            assert sms_response.http_status_code == 400
-            assert not sms_response.successful
+        assert sms_responses[0].message_id is None
+        assert sms_responses[0].http_status_code == 400
+        assert sms_responses[0].error_message == "Invalid To phone number format."
+        assert not sms_responses[0].successful
     
+    def test_send_sms_unauthorized_from_phone_number(self):
+
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+        
+        with pytest.raises(HttpResponseError) as ex:
+            # calling send() with sms values
+            sms_client.send(
+                from_="+18007342577",
```

Comment:
sorry somehow missed this, can we use fake ones here too?

Reply:
Done.